profiling/src/tracing_impl.rs

```diff
@@ -10,6 +10,13 @@ macro_rules! scope {
     };
 }
 
+// NOTE: Not supported as tracing does not support non literal spans. Use #[profiling::function] instead.
+#[macro_export]
+macro_rules! function_scope {
```

Comment:
Does the approach used by the superluminal backend not work here?

If this feature cannot be supported by this backend, then I do not want to add this feature to the crate. Imagine if some upstream crate chooses to use function_scope!() and it's not supported by the backend you chose. You wouldn't get any data, and that runs against the reason for this crate to exist: to let any library author add instrumented profiling to their crate, and for downstream users to be able to use any of the supported backends to see that data.

Reply:
I read more closely and see you mentioned tracing requires literals, not just &str. If there isn't a way of resolving this, then I think this would be problematic for adding this feature for the reason I mentioned above.

Reply:
Indeed tracing requires string literals, not just `&'static str`.  

>Imagine if some upstream crate chooses to use function_scope!() and it's not supported by the backend you chose.
It's true, but this seems to be a trade off. If the upstream crate chooses to not using profiling at all isn't it a much worse position to be in? (especially considering that the tracing backend is probably not the most use profiler here)

I did consider some options of what could be done to make the tracing crate supported:
* Use the human readable format string feature in tracing. It seems like tracing-tracy does not currently support this but I imagine it could be possible. Essentially we would do `event!(Level::INFO, "{}", function_name)`. 
* Add a special tag tracy.function_name to the event that tracing-tracy could read.

Reply:
tracing-tracy does support the human readable format argument, however we still need to provide a zone name, so this currently looks like follows. I think further refinement would involve changes to tracing-tracy.
![image](https://github.com/user-attachments/assets/c9cae630-c9ec-406c-a4f2-c6f5d6c0a336)

Reply:
I guess the comment "// NOTE: Not supported as tracing does not support non literal spans. Use #[profiling::function] instead." could be removed?